### PR TITLE
Expose useContractsOrError

### PR DESCRIPTION
**Pull request by paulmelnikow**
@qmaurmann @gmarceau

### CODE REVIEW

File: src/middleware.js [hunk 1]
```diff
@@ -8,4 +8,11 @@ cc.middleware = c.any.rename('middleware');
...
 
-module.exports.useContracts = cc.useContracts.wrap(impl.useContracts);
+// Reduce boilerplate for controllers of having to union with contracts.errorBody
+cc.useContractsOrError = c.fun({ requestContract: c.contract }, { responseBodyContract: c.contract })
+    .returns(cc.middleware);
+
+module.exports = {
+    useContracts: cc.useContracts.wrap(impl.useContracts),
+    useContractsOrError: cc.useContracts.wrap(impl.useContractsOrError),
+};
```
Comment (gmarceau) [inline src/middleware.js]: How about
```
exportContract = {
  useContracts: ...,
  useContractsOrError: ... 
};
module.exports = exportContract.wrap(require('./impl'))
```
?

**Comment by gmarceau** [comment]
👍  + left a comment

File: src/middleware.js [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
+        .returns(cc.middleware),
+};
+
+var wrapAll = function (contracts, impl) {
+    return _(contracts).mapObject(function (contract, key) {
+        return contract.wrap(impl[key]);
+    });
+};
+
+module.exports = wrapAll(middlewareContracts, require('./middleware.impl'));
```
Comment (paulmelnikow) [inline src/middleware.js]: @gmarceau Is this what you had in mind?

**Comment by qmaurmann** [comment]
Looks good, once the `wrapAll` issue is sorted out. Good to see that this stuff must be proving useful outside of `api-server`.

**Pull request merged**